app/src/main/java/com/example/drawingApp/DialogUtility.kt

```diff
@@ -0,0 +1,157 @@
+            dialog.dismiss()
+        }
+
+        view.findViewById<TextView>(R.id.newLayerOk).setOnClickListener {
+            val fieldString = layerField.text.toString().trim()
+            errorText.isVisible = fieldString.length > MAX_CHAR_LEN
+            if (!errorText.isVisible) {
+                if (!onDialogSubmission(fieldString) && !errorText.isVisible) {
+                    errorText.setText(R.string.layerExistsError)
+                    errorText.isVisible = !errorText.isVisible
+                } else {
+                    Toast.makeText(context, fieldString, Toast.LENGTH_SHORT)
+                        .show()
+                    dialog.dismiss()
+                }
+            }
+        }
+
+        dialog.setOnDismissListener { onDialogSubmission(null) }
+        dialog.show()
+        return dialog
+    }
+
+    /**
+     * opens a sheet dialog that allows the viewing of the layers in the list
+     * allows editing and deletion of items in the list
+     */
+    fun showViewingLayersDialog(
+        context: Context,
+        layerNames: List<String>,
+        /**
+         * called when the edit of a layer is clicked with the index of that layer
+         * ideally set the index to some variable to work with later
+         */
+        onEdit: (Int) -> Unit,
+        /**
+         * what happens when user clicks the delete button on a layer
+         * called in the Delete click
+         * remove the item at the index from the list your passing in
+         */
+        onDelete: (Int) -> Unit,
+        /**
+         * called on dismissals, do whatever tracking needed when the sheet closes
+         */
+        onSheetDismiss: () -> Unit
+    ): SheetObject {
+        val data = layerNames.map { layer -> LayerViewModel(layer) }.toMutableList()
+
+        lateinit var myAdapter: EditLayerAdapter
+        myAdapter = EditLayerAdapter(
+            data = data,
+            onDeleteClick = { layerModel ->
+                val index = data.indexOf(layerModel)
+                data.removeAt(index)
+                onDelete(index)
+                myAdapter.notifyItemRemoved(index)
+            },
+            onEditClick = { layerModel ->
+                val index = data.indexOf(layerModel)
+                onEdit(index)
+            }
+        )
+
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.edit_layer_recycler, null)
+        view.findViewById<RecyclerView>(R.id.editLayerRecycler).apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = myAdapter
+        }
+        view.findViewById<TextView>(R.id.closeLayerView).setOnClickListener { sheet.dismiss() }
+
+        sheet.setContentView(view)
+        sheet.setOnDismissListener { onSheetDismiss() }
+        sheet.show()
+
+        return SheetObject(onEdit = { index, str ->
```

Comment:
This could have the cleaner syntax if you made the suggestion from above

Reply:
updated